cdap-docs/admin-manual/source/operations/logging.rst

```diff
@@ -50,28 +102,409 @@ CDAP supports logging through standard `SLF4J (Simple Logging Facade for Java)
     output.emit(wordCount);
   }
 
-The log messages emitted by your application code can be viewed in different ways.
+This will emit "info" level messages from a flowlet when it is processing an event.
+
+Retrieving Log Messages from a Program
+--------------------------------------
+The log messages emitted by your application code can be retrieved by:
+
+- Using the :ref:`restful-api`: the :ref:`Logging HTTP RESTful API
+  <http-restful-api-logging>` details the available contexts that can be called to
+  retrieve different messages.
+- Log messages of a program can be viewed in the :ref:`CDAP UI <cdap-ui>`.
+
+Program Log File Locations
+--------------------------
+Program logs are stored in locations specified by properties in the
+:ref:`cdap-site.xml <appendix-cdap-default-logging>` file depending on the mode 
+of CDAP (Standalone or Distributed):
+
+- For **Standalone CDAP:** the property ``log.collection.root`` (default
+  ``${local.data.dir}/logs``) is the root location for collecting program logs when in
+  Standalone CDAP.
+
+- For **Distributed CDAP:** the property ``log.base.dir`` (default ``/logs/avro``) is the
+  base directory in HDFS for program logs
+
+.. _configuring-user-application-program-logging:
+
+Configuring Program Logs and Log Levels
+---------------------------------------
+The logging of an application's programs are configured by the ``logback-container.xml``
+file, packaged with the CDAP distribution. This "logback" does log rotation once a day at
+midnight and expires logs older than 14 days. Changes can be made to
+``logback-container.xml``; afterwards, the ``cdap-master`` process will need to be
+restarted for the modified logback file to take effect.
```

Comment:
It should also be mentioned that changing logback-container.xml will only affect programs that are started after the change; existing running programs will not be affected.